### PR TITLE
Remove `Publish with Confidence` promo dialog and related resources

### LINKED ISSUE #10474: Remove the "Publish with Confidence" dialog
I think we can remove the `Publish with Confidence` promo dialog, that was introduced long time ago to get users comfortable with the "async publishing" feature, and that's shown the first time the users publish a post from the mobile app - 
device-2019-09-02-170241

and fall back to the classic `Publish now` dialog, since the async publishing feature is not 2 years old?!
device-2019-09-02-170450

cc @iamthomasbishop

### PR DESCRIPTION
**Pull request by daniloercoli**
Fixes #10474 - Removes `Publish with Confidence` dialog + related resource.

To test:
- On a fresh install of the app
- Tap on the New Post button
- Write something and tap Publish
- The `Ready to Publish` dialog should be visible on the screen
Screenshot 2019-09-03 at 16 12 23

Tapping `Publish Now` the post is uploaded on the remote server, tapping `Keep Editing` it should not be uploaded.

Update release notes:

- [ x ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

cc @diegoreymendez - We should remove it from the iOS side as well

### CODE REVIEW

**Comment by jkmassel** [comment]
Howdy! The `13.3` code freeze is today, so I'm bumping this to `13.4`. If you'd like it to be part of the `13.3` release, please target the `release/13.3` branch, and DM me once it's merged so that I can include it as a new beta release!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1525,18 +1520,10 @@ private void primaryAction() {
...
-                }
+                showPublishConfirmationDialogAndPublishPost();
                 return;
             case SCHEDULE:
-                if (AppPrefs.isAsyncPromoRequired()) {
-                    showAsyncPromoDialog(mPost.isPage(), true);
-                } else {
-                    uploadPost(false);
-                }
+                uploadPost(false);
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: We can move this to the "other cases" below:

```java
            // In other cases, we'll upload the post without changing its status
            case SCHEDULE:
            case SUBMIT_FOR_REVIEW:
            case SAVE:
                uploadPost(false);
                break;
```

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -1835,16 +1814,6 @@ public void onPositiveClicked(@NonNull String instanceTag) {
 
     @Override
     public void onLinkClicked(@NonNull String instanceTag) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Better to remove that function completely, less lines in `EditPostActivity` = 🍰

**Comment by maxme** [comment]
Note: I merged `develop` into this branch (there was a conflict with some changes I made when I dropped the legacy and hybrid editors).

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 1 above)
Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Addressed in 0fafdc03907ecff2088d912ef368593e45832ce3

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 2 above)
Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Right call!

**Comment by daniloercoli** [comment]
This is now ready for another check. I've removed the unused code as per code review.

**Pull request merged**